bulk-model-sync-mps/src/main/kotlin/org/modelix/mps/model/sync/bulk/MPSBulkSynchronizer.kt

```diff
@@ -138,6 +148,71 @@ object MPSBulkSynchronizer {
         }
     }
 
+    @OptIn(ExperimentalSerializationApi::class)
+    @JvmStatic
+    fun importRepositoryFromModelServer() {
+        val repository = getRepository()
+
+        val includedModuleNames = parseRawPropertySet(System.getProperty("modelix.mps.model.sync.bulk.input.modules"))
+        val includedModulePrefixes = parseRawPropertySet(System.getProperty("modelix.mps.model.sync.bulk.input.modules.prefixes"))
+        val continueOnError = System.getProperty("modelix.mps.model.sync.bulk.input.continueOnError", "false").toBoolean()
+
+        val includedModulesFilter = IncludedModulesFilter(includedModuleNames, includedModulePrefixes)
+
+        val modelServerUrl = System.getProperty("modelix.mps.model.sync.bulk.server.url")
+        val repositoryId = RepositoryId(requireNotNull(System.getProperty("modelix.mps.model.sync.bulk.server.repository")) { "modelix.mps.model.sync.bulk.server.repository not specified" })
+        val branchName = System.getProperty("modelix.mps.model.sync.bulk.server.branch")
+        val branchRef = repositoryId.getBranchReference(branchName)
+        val versionHash: String? = System.getProperty("modelix.mps.model.sync.bulk.server.version.hash")
+        val baseVersionHash: String? = System.getProperty("modelix.mps.model.sync.bulk.server.version.base.hash")
+        val client = ModelClientV2.builder().url(modelServerUrl).build()
+        val version = runBlocking {
+            if (versionHash == null) client.lazyLoadVersion(branchRef) else client.lazyLoadVersion(repositoryId, versionHash)
+        }
+        val baseVersion = baseVersionHash?.let {
+            runBlocking {
+                client.lazyLoadVersion(repositoryId, it)
+            }
+        }
+        println("Loading version ${version.getContentHash()}")
+
+        val access = repository.modelAccess
+        access.executeCommandInEDT {
+            if (baseVersion == null) return@executeCommandInEDT
+
+            val invalidationTree = InvalidationTree(1_000_000)
+            val newTree = version.getTree()
+            newTree.visitChanges(
+                baseVersion.getTree(),
+                InvalidatingVisitor(newTree, invalidationTree),
+            )
+            val treePointer = TreePointer(newTree)
+            treePointer.runRead {
+                val synchronizer = ModelSynchronizer(
+                    CompositeFilter(listOf(includedModulesFilter, invalidationTree)),
+                    treePointer.getRootNode(),
+                    MPSRepositoryAsNode(repository),
+                    NodeAssociationToMps(MPSArea(repository)),
+                )
+                try {
+                    synchronizer.synchronize()
+                } catch (e: Exception) {
+                    // Exceptions are only visible in the MPS log file by default
+                    e.printStackTrace()
+                }
```

Comment:
Is that handling actually required. Can't `runRead` or the whole logic around automatically handle this?

Reply:
Yes, without it, the error would not be visible in the terminal and only in the MPS log file.

Reply:
Ok, an re-throwing is not required here?

Reply:
added re-throw